helloworld/helloworld.go

```diff
@@ -36,5 +77,7 @@ func Workflow(ctx workflow.Context, name string) (string, error) {
 func Activity(ctx context.Context, name string) (string, error) {
 	logger := activity.GetLogger(ctx)
 	logger.Info("Activity", "name", name)
-	return "Hello " + name + "!", nil
+
+	// return "Hello " + name + "!", nil
+	return "", temporal.NewApplicationError("some err", "NonRetryableError", NewNonRetryableErrorWithCode("foo msg", ErrorCodeFoo))
```

Comment:
here we explicitly wrap as a `temporal.ApplicationError`

Reply:
note the type param (`NonRetryableError`) which is how we specify this err is nonretryable